gobblin-metrics/src/main/java/gobblin/metrics/SerializedMetricReportReporter.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.metrics;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import org.apache.avro.io.Decoder;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.io.Encoder;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.base.Optional;
+
+
+/**
+ * Metric reporter that pushes serialized {@link gobblin.metrics.MetricReport}.
+ *
+ * The serialization format is defined by {@link #getEncoder}. By default, json encoder is used,
+ * but subclasses can override this method to use a different encoder.
+ *
+ * Concrete subclasses should implement {@link #pushSerializedReport}.
+ */
+public abstract class SerializedMetricReportReporter extends MetricReportReporter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SerializedMetricReportReporter.class);
+  public static final int SCHEMA_VERSION = 1;
+
+  private final Encoder encoder;
+  private final ByteArrayOutputStream byteArrayOutputStream;
+  private final DataOutputStream out;
+  protected final boolean reportMetrics;
+
+  private static Optional<SpecificDatumReader<MetricReport>> READER = Optional.absent();
```

Comment:
OK, it seems `deserializeReport` is not used anywhere in the same context, so can we move that to a utility class so we can get rid of this static variable?